src/titiler/core/titiler/core/factory.py

```diff
@@ -812,6 +824,299 @@ def wmts(
+                    
+            if out_wms_params.get("bbox"):
+
+                # WMS 1.3.0 is lame and flips the coords of EPSG:4326
+                if (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs")
+                    and out_wms_params.get("srs") == "EPSG:4326"
+                ):
+                    out_wms_params["bbox"] = [
+                        out_wms_params["bbox"][1],
+                        out_wms_params["bbox"][0],
+                        out_wms_params["bbox"][3],
+                        out_wms_params["bbox"][2],
+                    ]
+                    
+                # Overwrite CRS:84 with EPSG:4326 when specified
+                elif (
+                    out_wms_params.get("version") == "1.3.0"
+                    and out_wms_params.get("srs") == "CRS:84"
+                ):
+                    out_wms_params["srs"] = "EPSG:4326"
+                    
+                # `bounds` is compatible with reading from rio.io.reader
+                out_wms_params['bounds'] = out_wms_params['bbox']
+                    
+            print(out_wms_params)
+            return out_wms_params
+
+        @self.router.get("/wms")
+        def wms(
+            _request: Request,
+            request: WMSRequestType = "GetCapabilities",
+            version: WMSVersion = "1.1.1",
+            format: WMSMediaType = Query(
+                default=None, 
+                description="Output image type. Default is auto."
+            ),
+            bbox: Union[str, None] = Query(
+                default=None,
+                description="Bounding box for the WMS query. Should be in format of 'minx,miny,maxx,maxy'.",
+            ),
+            width: Union[int, None] = Query( 
+                default=None,
+                description="Width (in pixels) of returned GetMap image."
+            ),
+            height: Union[int, None]  = Query( 
+                default=None,
+                description="Height (in pixels) of returned GetMap image."
+            ),
+            srs: Union[EPSGCodes, None] = Query(
+                default=None,
+                description="Coordinate system of bbox as an EPSG code",
+            ),
+            # crs: Ignore if we stick with 
+            layers: list[str] = Query(
+                [""], 
+                description="COG dataset URL(s) or local path(s) to a COG"
+            ),
+            layer_params=Depends(self.layer_dependency),  # noqa
+            dataset_params=Depends(self.dataset_dependency),  # noqa
+            buffer: Optional[float] = Query(  # noqa
+                None,
+                gt=0,
+                title="Tile buffer.",
+                description="Buffer on each side of the given tile. It must be a multiple of `0.5`. Output **tilesize** will be expanded to `tilesize + 2 * buffer` (e.g 0.5 = 257x257, 1.0 = 258x258).",
+            ),
```

Comment:
buffer is only available for `.tile()` method

Reply:
Isn't `.part()` imported from here? https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/reader.py#L245

Reply:
🤦 true, I forgot about this